core/src/main/java/org/elasticsearch/index/query/ConstantScoreQueryBuilder.java

```diff
@@ -66,20 +70,78 @@ public ConstantScoreQueryBuilder boost(float boost) {
         return this;
     }
 
+    /**
+     * @return the boost factor
+     */
+    public float boost() {
+        return this.boost;
+    }
+
     @Override
     protected void doXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject(NAME);
         builder.field("filter");
-        filterBuilder.toXContent(builder, params);
+        queryBuilder.toXContent(builder, params);
+        builder.field("boost", boost);
+        builder.endObject();
+    }
+
+    @Override
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
+        Query innerFilter = queryBuilder.toQuery(parseContext);
+        if (innerFilter == null ) {
+            // no inner query, we ignore this and return no docs
+            return Queries.newMatchNoDocsQuery();
```

Comment:
I just talked to Adrien again and getting rid of the null-return value seems not to be so trivial after all. Take the example of a BoolQuery with a nested ConstantScoreQuery(). In case the later returns `matchNoDocsQuery` instead of null, it will make a difference whether it is nested under `should` or `must` (the former should not change the any results from other clauses, the later will return no docs). Formerly BoolQuery ignored any inner query parsed to null (which btw. I have to add to the toQuery() there as well)